hive-bigquery-connector-common/src/main/java/com/google/cloud/hive/bigquery/connector/output/direct/DirectRecordWriter.java

```diff
@@ -49,17 +48,19 @@ public class DirectRecordWriter
 
   JobConf jobConf;
   JobDetails jobDetails;
-  TaskAttemptID taskAttemptID;
   BigQueryDirectDataWriterHelper streamWriter;
   StructObjectInspector rowObjectInspector;
   Descriptors.Descriptor descriptor;
-  final String writerId;
+  final Path streamRefFile;
 
   public DirectRecordWriter(JobConf jobConf, JobDetails jobDetails) {
     this.jobConf = jobConf;
-    this.taskAttemptID = HiveUtils.taskAttemptIDWrapper(jobConf);
-    this.writerId = WriterRegistry.getWriterId();
     this.jobDetails = jobDetails;
+    String taskAttemptID = UUID.randomUUID().toString();
```

Comment:
we are relying on overwriting task attempt file when there are multiple attempts to handle attempt failures (and cleanup?). if each attempt has a different file, how is attempt faiilures handled?  also preserving the task attempt info helps troubleshooting.

Reply:
The reason I changed this is because there is no available hive query id in the Hadoop conf when using Spark SQL. Do you know how else we could handle this?

Reply:
@yigress Can you think of a way to reproduce this situation? It would be good to cover that in our test suite.

Reply:
the test would have first few attempts fail then another attempt succeed. I can't think of a good way to do that. maybe another thread monitor the writing location and switching permissions between first and later attempts, or something similar?

Reply:
Note: The Spark task ID now uses the Spark app ID and the partition ID. The temp output folder also uses the ".hive-staging-XXX" folder.